api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/EriDSDCohortQueries.java

```diff
@@ -604,30 +604,38 @@ public CohortDefinition getPatientsWhoAreActiveWithNextPickupAs3Months() {
     CohortDefinition patientsScheduled =
         getPatientsScheduled(
             hivMetadata.getReturnVisitDateForArvDrugConcept(),
-            Arrays.asList(hivMetadata.getARVPharmaciaEncounterType()),
+            Arrays.asList(hivMetadata.getARVPharmaciaEncounterType(),hivMetadata.getMasterCardDrugPickupEncounterType()),
             97,
             83);
     CohortDefinition quarterly = getPatientsWithQuarterlyTypeOfDispensation();
     CohortDefinition startOrContinue = getPatientsWithStartOrContinueOnQuarterlyDispensation();
     CohortDefinition completed = getPatientsWithCompletedOnQuarterlyDispensation();
+    CohortDefinition pregnant =  txNewCohortQueries.getPatientsPregnantEnrolledOnART();
+    CohortDefinition breastFeeding =  txNewCohortQueries.getTxNewBreastfeedingComposition();
+    CohortDefinition patientsWithTB =  tbCohortQueries.getPatientsOnTbTreatment();
 
     String mappings = "onOrBefore=${endDate},location=${location}";
     String dispensationMappings =
         "onOrAfter=${startDate},onOrBefore=${endDate},locationList=${location}";
+    String pregnantMappings = "startDate=${endDate-9m},endDate=${endDate},location=${location}";
+    String breastFeedingMappings = "onOrAfter=${endDate-18m},onOrBefore=${endDate},location=${location}";
+    String tbMappinggs = "startDate=${startDate},endDate=${endDate},location=${location}";
 
     cd.addSearch("TxCurr", EptsReportUtils.map(txCurr, mappings));
     cd.addSearch("scheduled", EptsReportUtils.map(patientsScheduled, mappings));
     cd.addSearch("quarterly", EptsReportUtils.map(quarterly, dispensationMappings));
     cd.addSearch("startOrContinue", EptsReportUtils.map(startOrContinue, dispensationMappings));
     cd.addSearch("completed", EptsReportUtils.map(completed, dispensationMappings));
-
-    cd.setCompositionString("TxCurr AND (scheduled OR quarterly OR startOrContinue) NOT completed");
+    cd.addSearch("pregnant", EptsReportUtils.map(pregnant,pregnantMappings));
+    cd.addSearch("breastfeeding", EptsReportUtils.map(breastFeeding,breastFeedingMappings));
+    cd.addSearch("tbPatients", EptsReportUtils.map(patientsWithTB,tbMappinggs));
+    cd.setCompositionString("TxCurr AND (scheduled OR quarterly OR startOrContinue) NOT (completed OR pregnant OR breastfeeding OR tbPatients");
```

Comment:
```suggestion
    cd.addSearch("PregnantAndBreastfeedingAndOnTBTreatment", EptsReportUtils.map(getPregnantAndBreastfeedingAndOnTBTreatment(),
    "endDate=${endDate},location=${location}"));
    cd.setCompositionString("TxCurr AND (scheduled OR quarterly OR startOrContinue) NOT (completed OR PregnantAndBreastfeedingAndOnTBTreatment");
```

Reply:
@Edgencio please have a look

Reply:
Thanks for the suggestions @psmatsinhe !